dlc/src/digit_trie.rs

```diff
@@ -0,0 +1,637 @@
+//! Data structure to store and lookup digit decomposition data.
+//!
+
+use trie::{LookupError, LookupResult, Node};
+
+/// Structure to store data inserted and looked-up based on digit paths.
+pub struct DigitTrie<T> {
+    /// Use the arena allocated approach which makes it easier to
+    /// satisfy the borrow checker.  
+    store: Vec<Node<DigitLeaf<T>, DigitNode<T>>>,
+    root: Option<usize>,
+    base: usize,
+}
+
+/// Structure used to iterated through a `DigitTrie` values.
+pub struct DigitTrieIter<'a, T> {
+    trie: &'a DigitTrie<T>,
+    index_stack: Vec<(Option<usize>, isize)>,
```

Comment:
doc comment for `index_stack` would be helpful

Reply:
Added as well as a bit more info about the type of traversal performed by the iterator.